src/main/java/org/broadinstitute/hellbender/engine/filters/UMIReadFilter.java

```diff
@@ -0,0 +1,42 @@
+package org.broadinstitute.hellbender.engine.filters;
+
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.hellbender.cmdline.ReadFilterArgumentDefinitions;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
+public class UMIReadFilter extends ReadFilter {
+    static final long serialVersionUID = 1L;
+    public static final String UMI_TAG = "RX";
+
+    @Argument(fullName= ReadFilterArgumentDefinitions.UMI_NAME, doc = "umi must have the format 'XXX-XXX'")
+    public String umi;
```

Comment:
Should we validate that constraint somewhere?  @cmnbroad is there a way to do validation of the arguments in a readfilter?

Reply:
Nothing built it. Maybe `UMI` should have a `String` constructor that does the splitting/parsing.